Tribler/community/tunnel/hidden_community.py

```diff
@@ -312,26 +397,34 @@ def on_key_request(self, messages):
 
     def check_key_response(self, messages):
         for message in messages:
-            if not message.source.startswith(u"circuit_"):
```

Comment:
I verified this yesterday, and got the insight that the introduction point does an exit_data with the dispersy key-response directly to the downloader, no circuits were involved there. Downloading over hidden services therefore only worked if the candidates of the downloader and intropoint already knew each other or when no firewall/nat is between them.